From 7d2bd264b6025e84f983de5c83f79db372c93fac Mon Sep 17 00:00:00 2001
From: Teo Mrnjavac <teo@kde.org>
Date: Tue, 17 Jan 2017 18:03:51 +0100
Subject: [PATCH] Useless check is useless.

---
 src/modules/bootloader/main.py | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/modules/bootloader/main.py b/src/modules/bootloader/main.py
index 9feb02e..786342b 100644
--- a/src/modules/bootloader/main.py
+++ b/src/modules/bootloader/main.py
@@ -282,8 +282,6 @@ def run():
 
     :return:
     """
-    if libcalamares.globalstorage.value("bootLoader") is None:
-        return None
 
     fw_type = libcalamares.globalstorage.value("firmwareType")
     prepare_bootloader(fw_type)
