From d99d7bfdff49e16065284c3407312c13ca21901f Mon Sep 17 00:00:00 2001
From: Teo Mrnjavac <teo@kde.org>
Date: Fri, 19 Aug 2016 12:48:14 +0200
Subject: [PATCH] Expose CalamaresUtils::obscure as libcalamares.utils.obscure.

---
 src/libcalamares/PythonJob.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/libcalamares/PythonJob.cpp b/src/libcalamares/PythonJob.cpp
index d6913ec..e009726 100644
--- a/src/libcalamares/PythonJob.cpp
+++ b/src/libcalamares/PythonJob.cpp
@@ -206,6 +206,15 @@ BOOST_PYTHON_MODULE( libcalamares )
             "subprocess.CalledProcessError if something went wrong."
         )
     );
+    bp::def(
+        "obscure",
+        &CalamaresPython::obscure,
+        bp::args( "s" ),
+        "Simple string obfuscation function based on KStringHandler::obscure.\n"
+        "Returns a string, generated using a simple symmetric encryption.\n"
+        "Applying the function to a string obscured by this function will result "
+        "in the original string."
+    );
 }
 
 
